networks.go

```diff
@@ -0,0 +1,74 @@
+package unifi
+
+import (
+	"encoding/json"
+	"fmt"
+)
+
+// GetNetworks returns a response full of network data from the UniFi Controller.
+func (u *Unifi) GetNetworks(sites []*Site) ([]Network, error) {
+	networks := make([]Network, 0)
+
+	for _, site := range sites {
+		var response struct {
+			Data []json.RawMessage `json:"data"`
+		}
+
+		networkPath := fmt.Sprintf(APINetworkPath, site.Name)
+		if err := u.GetData(networkPath, &response); err != nil {
+			return nil, err
+		}
+
+		for _, data := range response.Data {
+			network := u.parseNetwork(data, site.SiteName)
+			networks = append(networks, *network)
+		}
+	}
+
+	return networks, nil
+}
+
+// parseNetwork parses the raw JSON from the Unifi Controller into network structures.
+func (u *Unifi) parseNetwork(data json.RawMessage, siteName string) *Network {
+	network := new(Network)
+	u.unmarshalNetwork(data, &network)
+	return network
+}
+
+// unmarshalNetwork handles logging for the unmarshal operations in parseNetwork().
+func (u *Unifi) unmarshalNetwork(data json.RawMessage, v interface{}) (err error) {
+	if err = json.Unmarshal(data, v); err != nil {
+		u.ErrorLog("json.Unmarshal(): %v", err)
+		u.ErrorLog("Enable Debug Logging to output the failed payload.")
+
+		json, err := data.MarshalJSON()
+		u.DebugLog("Failed Payload: %s (marshal err: %v)", json, err)
+		u.DebugLog("The above payload can prove useful during torubleshooting when you open an Issue:")
+		u.DebugLog("==- https://github.com/unifi-poller/unifi/issues/new -==")
+	}
+
+	return err
+}
+
```

Comment:
This doesn't seem like the right pattern for this request, but if you're going to do it, you can just re-use this method: https://github.com/unifi-poller/unifi/blob/master/devices.go#L84 (they're identical, so no need to re-write it).
```suggestion
```
The method above has a suggestion to make the change.

Reply:
yeah reusing the other method seems better.  this is the first i've done anything with json / (un)marshaling in Go, not sure what the best design would be.  happy to take whatever suggestions you have.